functions/src/export-csv.ts

```diff
@@ -145,130 +170,149 @@ function writeRow(
+  if (!loi.geometry) {
+    console.debug(`Skipping LOI ${loi.id} - missing geometry`);
+    return;
+  }
+  row.push(quote(toWkt(loi.geometry)));
   // Header: One column for each loi property (merged over all properties across all LOIs)
-  row.push(...getPropertiesByName(loiDoc, loiProperties));
-  // TODO(#1288): Clean up remaining references to old responses field
-  const data =
-    submission['data'] ||
-    submission['responses'] ||
-    submission['results'] ||
-    {};
+  getPropertiesByName(loi, loiProperties).forEach(v => row.push(quote(v)));
+  const data = submission.taskData;
   // Header: One column for each task
-  tasks.forEach((task, taskId) => row.push(getValue(taskId, task, data)));
+  tasks.forEach((task, taskId) =>
+    row.push(quote(getValue(taskId, task, data)))
+  );
   // Header: contributor_username, contributor_email
-  const contributor = submission['created']
-    ? (submission['created'] as Dict)['user']
-    : [];
-  row.push(contributor['displayName'] || '');
-  row.push(contributor['email'] || '');
+  row.push(quote(submission.created?.displayName));
+  row.push(quote(submission.created?.emailAddress));
   csvStream.write(row);
 }
 
-/**
- * Returns the WKT string converted from the given geometry object
- *
- * @param geometryObject - the GeoJSON geometry object extracted from the LOI. This should have format:
- *   {
- *      coordinates: any[],
- *      type: string
- *   }
- * @returns The WKT string version of the object
- * https://www.vertica.com/docs/9.3.x/HTML/Content/Authoring/AnalyzingData/Geospatial/Spatial_Definitions/WellknownTextWKT.htm
- *
- * @beta
- */
-function toWkt(geometryObject: any): string {
-  return geojsonToWKT(Datastore.fromFirestoreMap(geometryObject));
+function quote(value: any): string {
+  if (value === null || value === undefined) {
+    return '';
+  }
+  if (typeof value === 'number') {
+    return value.toString();
+  }
+  const escaped = value.toString().replaceAll('"', '""');
+  return '"' + escaped + '"';
+}
+
+function toWkt(geometry: Pb.IGeometry): string {
+  return geojsonToWKT(toGeoJsonGeometry(geometry));
 }
 
 /**
- * Returns the string representation of a specific task element result.
+ * Returns the string or number representation of a specific task element result.
  */
-function getValue(taskId: string, task: Task, data: any) {
-  const result = data[taskId] || '';
-  if (
-    task.type === 'multiple_choice' &&
-    Array.isArray(result) &&
-    task.options
-  ) {
-    return result.map(id => getMultipleChoiceValues(id, task)).join(', ');
-  } else if (task.type === 'capture_location') {
-    if (!result) {
-      return '';
-    }
-    return toWkt(result.geometry || result);
+function getValue(
+  taskId: string,
+  task: Task,
+  data: Pb.ITaskData[]
+): string | number | null {
+  const result = data.find(d => d.taskId === taskId);
+  if (!result) {
+    return null;
+  }
+  if (result.textResponse) {
+    return result.textResponse.text || null;
```

Comment:
Better to use `??` to allow for empty string response.

Reply:
Done.